tester/src/test/scala/spinal/tester/scalatest/FormalArbiterTester.scala

```diff
@@ -0,0 +1,455 @@
+
+        val select = UInt(log2Up(portCount) bit)
+        val selectOH = Bits(portCount bit)
+        val inputs = Vec(slave(Stream(dataType)), portCount)
+        val output = master(Stream(dataType))
+
+        select := dut.io.chosen
+        selectOH := dut.io.chosenOH
+        output << dut.io.output
+
+        when(reset || past(reset)) {
+          for (i <- 0 until portCount) {
+            assume(inputs(i).valid === False)
+          }
+        }
+
+        for (i <- 0 until portCount) {
+          inputs(i) >> dut.io.inputs(i)
+          inputs(i).formalAssumesSlave()
+        }
+        output.formalCovers(2)
+
+        assert(select < portCount)
+        assert(select === OHToUInt(selectOH))
+        assert(output.payload === inputs(select).payload)
+
+        val allInputsValidOH = OHMasking.first(Vec(inputs.map(_.valid))).asBits
+        assert(selectOH === allInputsValidOH)
+        val stallStableSel = output.isStall && stable(allInputsValidOH)
+        cover(stallStableSel)
+        when(stallStableSel) {
+          output.formalAssertsMaster()
+        }
+      })
+  }
+
+  test("Arbiter-lowerfirst-transactionLock-verify") {
+    FormalConfig
+      .withBMC(20)
+      .withProve(20)
+      .withCover(20)
+      // .withDebug
+      .doVerify(new Component {
+        val portCount = 5
+        val dataType = Bits(8 bits)
+
+        val dut = FormalDut(
+          new StreamArbiter(dataType, portCount)(
+            StreamArbiter.Arbitration.lowerFirst,
+            StreamArbiter.Lock.transactionLock
+          )
+        )
+        val reset = ClockDomain.current.isResetActive
+
+        assumeInitial(reset)
+
+        val select = UInt(log2Up(portCount) bit)
+        val selectOH = Bits(portCount bit)
+        val inputs = Vec(slave(Stream(dataType)), portCount)
+        val output = master(Stream(dataType))
+
+        select := dut.io.chosen
+        selectOH := dut.io.chosenOH
+        output << dut.io.output
+
+        when(reset || past(reset)) {
+          for (i <- 0 until portCount) {
+            assume(inputs(i).valid === False)
+            assert(!dut.locked)
+          }
+        }
+
+        val notReset = !(reset || past(reset))
+        cover(notReset)
+        when(notReset) {
+          assert(past(output.isStall) === dut.locked)
+        }
+
+        for (i <- 0 until portCount) {
+          inputs(i) >> dut.io.inputs(i)
+          inputs(i).formalAssumesSlave()
+        }
+        output.formalCovers(2)
+        output.formalAssertsMaster()
+
+        assert(select < portCount)
+        assert(select === OHToUInt(selectOH))
+        assert(output.payload === inputs(select).payload)
+        assert(output.valid === inputs(select).valid)
+
+        val shouldStableSelCond = past(output.isStall) && notReset
+        cover(shouldStableSelCond)
+        when(shouldStableSelCond) {
+          assert(stable(select))
+        }
+
+        val allInputsValidOH = OHMasking.first(Vec(inputs.map(_.valid))).asBits
```

Comment:
How about put these code into a function called verifyLowFirst() and call it in similar test functions.